src/material/list/selection-list.ts

```diff
@@ -579,7 +579,9 @@ export class MatSelectionList extends _MatSelectionListMixinBase implements CanD
 
   /** Returns the values of the selected options. */
   private _getSelectedOptionValues(): string[] {
-    return this.options.filter(option => option.selected).map(option => option.value);
+    return this.options.filter(option => {
+      return option.selected && option.value !== undefined;
+    }).map(option => option.value);
```

Comment:
doesn't this cause semantic changes to the NgModel value of the `MatSelectionList`? i.e. previously the form control value could be `[undefined]`. I could see the argument of saying that `undefined` can never be a value due limitations of how selection-list is implemented. Though we should capture the intended behavior in a test?

Reply:
You're right, but I'm not sure whether there's a cleaner alternative. We could also have a property that keeps track of whether the `value` setter was ever invoked and gate it based on that, but it feels very hacky.

Reply:
Yeah not sure either if there is a cleaner way. I have the feeling that there should be no special handling for values like `undefined`. i.e. if multiple options have the same _value_, it's expected that all matching ones are selected. Though it feels contradictory to the fact that individual options have a `selected` input.

Reply:
Agreed, but we wrote ourselves into a corner with the `selected` input.

Reply:
Yeah 😄 I'm not sure how to proceed here. maybe discuss in the next team meeting?

Reply:
Sure, I'll add it to the agenda.

Reply:
This is ancient, but any memory of whether we talked about this and what we decided?